disnake/ui/action_row.py

```diff
@@ -264,19 +351,131 @@ def add_text_input(
                 required=required,
                 min_length=min_length,
                 max_length=max_length,
-            )
+            ),
+        )
+
+    def remove_item(self, item: UIComponentT) -> None:
+        """Remove a component from the action row.
+
+        Parameters
+        ----------
+        item: :class:`WrappedComponent`
+            The component to remove from the action row.
+
+        Raises
+        ------
+        ValueError:
+            The component could not be found on the action row.
+        """
+        self._children.remove(item)
+
+    def pop(self, index: int) -> UIComponentT:
+        """Pop the component at the provided index from the action row.
+        This takes into account the width of components.
+
+        Parameters
+        ----------
+        index: :class:`int`
+            The index at which to pop the component.
+
+        Raises
+        ------
+        IndexError
+            There is no component at the provided index.
+        """
+        self.remove_item(component := self[index])
+        return component
+
+    @property
+    def _underlying(self) -> ActionRowComponent:
+        return ActionRowComponent._raw_construct(
+            type=self.type,
+            children=[comp._underlying for comp in self._children],
         )
 
     def to_component_dict(self) -> ActionRowPayload:
         return self._underlying.to_dict()
 
+    def __delitem__(self, index: int) -> None:
+        del self._children[index]
+
+    def __getitem__(self, index: int) -> UIComponentT:
+        aggregate = 0
+        for component in self._children:
+            aggregate += component.width
+            if aggregate >= index:
+                return component
+        raise IndexError("ActionRow index out of range")
+
+    @classmethod
+    def with_modal_components(cls) -> ActionRow[ModalUIComponent]:
+        """Create an empty action row meant to store components compatible with
+        :class:`disnake.ui.Modal`. Saves the need to import type specifiers to
+        typehint empty action rows.
+
+        .. versionadded:: 2.6
+
+        Returns
+        -------
+        :class:`ActionRow`:
+            The newly created empty action row, intended for modal components.
+        """
+        return ActionRow[ModalUIComponent]()
+
+    @classmethod
+    def with_message_components(cls) -> ActionRow[MessageUIComponent]:
+        """Create an empty action row meant to store components compatible with
+        :class:`disnake.Message`. Saves the need to import type specifiers to
+        typehint empty action rows.
+
+        .. versionadded:: 2.6
+
+        Returns
+        -------
+        :class:`ActionRow`:
+            The newly created empty action row, intended for message components.
+        """
+        return ActionRow[MessageUIComponent]()
+
+    @classmethod
+    def from_message(cls, message: Message) -> List[ActionRow[MessageUIComponent]]:
```

Comment:
this needs to always return an actionrow, no lists. Its also fine if it returns an actionrow which contains action rows, but we should always return an actionrow.

Reply:
Update: I have looked at the discord docs and this is accurate, but it should perhaps have a different name as it doesn't *actually* return an ActionRow. Not sure what that name would be...

Reply:
For now, I renamed this to `ActionRow.rows_from_message` to at least somewhat indicate the fact that it will return multiple rows. However, the name's still very much up for debate.

Reply:
Accidently started a new thread, respond here: https://github.com/DisnakeDev/disnake/pull/462#discussion_r885306985